scripts/lib/downloadImages.ts

```diff
@@ -17,23 +17,37 @@ import { createWriteStream } from "node:fs";
 import { finished } from "stream/promises";
 import { Readable } from "stream";
 import pMap from "p-map";
+import { startWebServer, closeWebServer } from "../lib/webServer";
 
 export async function downloadImages(
   images: Array<{ src: string; dest: string }>,
+  imagesPath: string,
 ) {
-  await pMap(
-    images,
-    async (img) => {
-      if (await pathExists(img.dest)) return;
-      const response = await fetch(img.src);
-      if (response.ok) {
-        await mkdirp(dirname(img.dest));
-        const stream = createWriteStream(img.dest);
-        await finished(Readable.fromWeb(response.body as any).pipe(stream));
-      } else {
-        console.log(`Error downloading ${img.src} to ${img.dest}`);
-      }
-    },
-    { concurrency: 4 },
+  const missingImages = images.filter(
+    async (img) => !(await pathExists(img.dest)),
   );
+
+  if (missingImages.length == 0) {
+    return;
+  }
+
+  await startWebServer(`${imagesPath}/artifact`, 8000);
```

Comment:
The 8000 is a magic number. How about we remove from `startWebServer` and `closeWebServer` this argument and hardcode it? In the web server file, you can set a constant `PORT = 8000`.